tests/test-solve.C

```diff
@@ -0,0 +1,195 @@
+
+	while( m!=FFPACK::Rank( F, m, m, A, lda) ){
+		RandomMatrix (F, m, m, A, lda, Rand);
+	}	
+	RandomMatrix (F, m, 1, B, incb, Rand);
+	RandomMatrix (F, m, 1, B2, incb, Rand);
+
+	FFLAS::fassign (F, m, 1, B, incb, B2, incb);
+	FFLAS::fassign (F, m, m, A, lda, A2, lda);
+	FFLAS::WriteMatrix(std::cout<<"b:="<<std::endl,F,m,1,B,incb)<<std::endl;
+
+	FFLAS::Timer t; t.clear();
+	double time=0.0;
+	t.clear();
+	t.start();
+	FFPACK::Solve(F, m, A2, lda, x, incx, B2, incb);
+	t.stop();
+	time+=t.usertime();
+
+	FFLAS::fgemv(F, FFLAS::FflasNoTrans, m, m, F.one, A, lda, x, incx, F.zero, B2, incb);
+
+	bool ok = true;
+	if (FFLAS::fequal (F, m, 1, B2, incb, B, incb)){
+
+		cout << "PASSED ("<<time<<")"<<endl;
+
+	} else{
+		FFLAS::WriteMatrix(std::cout<<"A*x:="<<std::endl,F,m,1,B2,incb)<<std::endl;
+		FFLAS::WriteMatrix(std::cout<<"b:="<<std::endl,F,m,1,B,incb)<<std::endl;
+		cout << "FAILED ("<<time<<")"<<endl;
+		ok=false;
+		
+	}
+	
+	FFLAS::fflas_delete(A);
+	FFLAS::fflas_delete(A2);
+	FFLAS::fflas_delete(B);
+	FFLAS::fflas_delete(B2);
+	FFLAS::fflas_delete(x);
+	return ok;
+}
+template <class Field>
+bool run_with_field (Givaro::Integer q, size_t b, size_t m, size_t iters, uint64_t seed){
+	bool ok = true ;
+	int nbit=(int)iters;
+
+	while (ok &&  nbit){
+		//typedef typename Field::Element Element ;
+		// choose Field
+		Field* F= chooseField<Field>(q,b);
+		typename Field::RandIter G(*F,0,seed);
+		if (F==nullptr)
+			return true;
+
+		cout<<"Checking with ";F->write(cout)<<endl;
+
+		ok = ok && check_solve(*F,m,G);
+
+		nbit--;
+		delete F;
+	}
+	
+	return ok;
+}
+
+int main(int argc, char** argv)
+{
+	cerr<<setprecision(10);
+	Givaro::Integer q=-1;
+	size_t b=0;
+	size_t m=6;
+
+	size_t iters=100;
+	bool loop=false;
+	uint64_t seed = time(NULL);
+	Argument as[] = {
+		{ 'q', "-q Q", "Set the field characteristic (-1 for random).",         TYPE_INTEGER , &q },
+		{ 'b', "-b B", "Set the bitsize of the field characteristic.",  TYPE_INT , &b },
+		{ 'm', "-m M", "Set the dimension of unknown square matrix.",      TYPE_INT , &m },
+
+		{ 'i', "-i R", "Set number of repetitions.",            TYPE_INT , &iters },
+		{ 'l', "-loop Y/N", "run the test in an infinite loop.", TYPE_BOOL , &loop },
+		{ 's', "-s seed", "Set seed for the random generator", TYPE_INT, &seed },
+                END_OF_ARGUMENTS
+        };
+
+	FFLAS::parseArguments(argc,argv,as);
+
+	bool ok = true;
+
+	do{
+		ok = ok && run_with_field<Modular<double> >(q,b,m,iters,seed);
+
+		ok = ok && run_with_field<ModularBalanced<double> >(q,b,m,iters,seed);
+
+		ok = ok && run_with_field<Modular<float> >(q,b,m,iters,seed);
+
+		ok = ok && run_with_field<ModularBalanced<float> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<Modular<int32_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<ModularBalanced<int32_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<Modular<int64_t> >(q,b,m,iters,seed);
+		ok = ok && run_with_field<ModularBalanced<int64_t> >(q,b,m,iters,seed);
+
```

Comment:
missing specializations for Modular<Givaro::Integer> with fixed bitsize ( 5) and with bit size set to b. See test-lu.C for example.